File: 4/src/main/java/com/pjwstk/Calculator.java
package com.pjwstk;

public class Calculator {

    public Calculator(){}

    public double add(double a, double b) {
        return a + b;
    }

    public double subtract(double a, double b) {
        return a - b;
    }

    public double multiply(double a, double b) {
        return a * b;
    }

    public double divide(double a, double b) throws ArithmeticException {
        if (b == 0) throw new ArithmeticException("You can't divide by 0");
        return a / b;
    }
    public String stringifyAddition(double number){
        return String.valueOf(number);
    }
    public boolean isOdd(int number){
        return number % 2 == 1;
    }
}

File: 7/calculator.py

class Calculator:

    def add(self, a, b):
        return a + b
    
    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        return a / b

class Table:

    def has_number(self, array, x):
        return x in array

    def sum(self, array):
        result = 0
        for number in array:
            result += number
        return result
